From aba9904a3d6ce898a2881fb30ea7252fee119ea4 Mon Sep 17 00:00:00 2001
From: Jonas Larsson <ljonas@google.com>
Date: Mon, 8 Oct 2018 16:58:19 -0700
Subject: [PATCH 1/1] Silence NXP stdout prints

Always printing VPU lib version information to stdout breaks use
cases where a user want to use stdout for something else, like
piping video data.
Instead use GStreamer logging facilities.

Change-Id: I993cfc87999302a6714769a80b022310c897efb8
---
 plugins/vpu/gstvpudecobject.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/plugins/vpu/gstvpudecobject.c b/plugins/vpu/gstvpudecobject.c
index a2d71eb..41ccc0a 100644
--- a/plugins/vpu/gstvpudecobject.c
+++ b/plugins/vpu/gstvpudecobject.c
@@ -318,11 +318,11 @@ gst_vpu_dec_object_start (GstVpuDecObject * vpu_dec_object)
         gst_vpu_dec_object_strerror(ret));
   }
 
-  g_print("====== VPUDEC: %s build on %s %s. ======\n",  (VERSION),__DATE__,__TIME__);;
-  g_print("\twrapper: %d.%d.%d (%s)\n", wrapper_version.nMajor, wrapper_version.nMinor, 
-    wrapper_version.nRelease, (wrapper_version.pBinary? wrapper_version.pBinary:"unknow"));
-  g_print("\tvpulib: %d.%d.%d\n", version.nLibMajor, version.nLibMinor, version.nLibRelease);
-  g_print("\tfirmware: %d.%d.%d.%d\n", version.nFwMajor, version.nFwMinor, version.nFwRelease, version.nFwCode);
+  GST_INFO_OBJECT(vpu_dec_object, "====== VPUDEC: %s build on %s %s. ======",  (VERSION),__DATE__,__TIME__);
+  GST_INFO_OBJECT(vpu_dec_object, "wrapper: %d.%d.%d (%s)", wrapper_version.nMajor, wrapper_version.nMinor,
+      wrapper_version.nRelease, (wrapper_version.pBinary ? wrapper_version.pBinary : "unknown"));
+  GST_INFO_OBJECT(vpu_dec_object, "vpulib: %d.%d.%d", version.nLibMajor, version.nLibMinor, version.nLibRelease);
+  GST_INFO_OBJECT(vpu_dec_object, "firmware: %d.%d.%d.%d", version.nFwMajor, version.nFwMinor, version.nFwRelease, version.nFwCode);
 
   /* mem_info contains information about how to set up memory blocks
    * the VPU uses as temporary storage (they are "work buffers") */
-- 
2.26.0

